Add week 1 written problems

jlperona opened the pull request:
It's hard for me to know exactly what you've covered through your lecture notes alone, so I'm left somewhat guessing. It's also hard to come up with really good problems for this topic in particular... I feel like I'll have an easier time as we get closer to the single-cycle and performance stuff.

Feel free to suggest edits for me to make or edit it yourself in the branch.

Commit:
Add week 1 written problems.

It's hard to come up with these...

powerjg commented on written/week1.md:
```diff
@@ -0,0 +1,30 @@
+
+## Problems
+
+1. What is the Iron Law?
```
This is going to be covered in Performance Lecture, ~#8 I think.

jlperona replied:
Hmm, it was covered in the Turing Lecture too. Is that fair game, or would you rather wait until you officially cover it?

powerjg reviewed (commented):
Could you add a couple of questions related to security being a first-order design constraint? Something about, how do we decide if a system is "good". Is it performance, energy, power, security, all of the above? etc. The first lecture was all on security, spectre, and meltdown.

Key things from lecture 2:
- Reliability going down since transistors are getting smaller
- New underlying technology changes the architecture design (e.g., CMOS, end of Moore's law, end of Dennard scaling, better compilers, etc.)

Lecture 3: 
- Dynamic vs static power/energy
- voltage is the dominate term (v^2)

Another point I harped on:
- What is the ISA?
  - Answer: The *interface* between software and hardware
- What is RISC-V
  - Just an *interface*, not an implementation.
- Can ask the same thing about x86, ARM, etc.

Thanks for getting this started! I know it's hard to try to read my mind! I'll share my one note notebook that has all of my personal lecture notes with you. That should help.

powerjg commented:
Sure, you can put it in now :D. Might as well show it to them many times!

On Thu, Jan 17, 2019 at 4:15 PM jlperona <notifications@github.com> wrote:

> *@jlperona* commented on this pull request.
> ------------------------------
>
> In written/week1.md
> <https://github.com/jlpteaching/ECS154B/pull/29#discussion_r248889223>:
>
> > +3. [More technology](https://github.com/jlpteaching/ECS154B/blob/master/lecture%20notes/Lecture-3.pdf)
> +
> +## Topics covered
> +
> +* Computer architecture overview
> +* History of computing
> +  * Moore's law
> +  * Dennard scaling
> +* CMOS devices
> +  * Energy
> +  * Power
> +* Trends in computing
> +
> +## Problems
> +
> +1. What is the Iron Law?
>
> Hmm, it was covered in the Turing Lecture too. Is that fair game, or would
> you rather wait until you officially cover it?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/jlpteaching/ECS154B/pull/29#discussion_r248889223>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAGsHNvhk1NGAI22cf63ghGyy5v_5oL4ks5vERIYgaJpZM4aE8Zf>
> .
>


-- 
Jason Lowe-Power
Assistant Professor, Computer Science Department
University of California, Davis
3049 Kemper Hall
https://arch.cs.ucdavis.edu/

Commit:
Finish week 1 written homework, add to week 2

Made some modifications and moved some things around based on your note
s. I'll try to finish up week 2 tomorrow.

powerjg commented:
Looks good. Thanks!
